iOS/DuckDuckGo/MainViewController+Segues.swift

```diff
@@ -225,25 +225,25 @@ extension MainViewController {
     func segueToDuckDuckGoSubscription() {
         Logger.lifecycle.debug(#function)
         hideAllHighlightsIfNeeded()
-        launchSettings {
+        launchSettings(completion: {
             $0.triggerDeepLinkNavigation(to: .subscriptionFlow())
-        }
+        }, deepLinkTarget: .subscriptionFlow())
```

Comment:
This looks redundant, but it's to handle two different cases:
1. When the deep link causes settings to be displayed for the first time, the `deepLinkTarget` parameter is used to show the flow
2. When settings is already visible, settings has already been launched so `deepLinkTarget` does nothing - instead, that is where the original `completion` block comes in, as it pushes the deep linked VC onto the stack

It's pretty gross, but it's how the API works and I'd love to avoid meaningful refactoring if possible 🤷